data_management_utils/utils.py

```diff
@@ -6,9 +6,89 @@
 from google.api_core.exceptions import NotFound
 import apache_beam as beam
 import zarr
+import xarray as xr
 import datetime
 from dataclasses import dataclass
 
+
+# ----------------------------------------------------------------
+# ----------------- Functions -----------------------------------
+# ----------------------------------------------------------------
+# wrapper functions (not sure if this works instead of the repeated copy and paste in the transform below)
+def log_to_bq(iid: str, store: zarr.storage.FSStore, table_id: str):
+    bq_interface = BQInterface(table_id=table_id)
+    iid_entry = IIDEntry(iid=iid, store=store.path)
+    bq_interface.insert(iid_entry)
+
+
+# ----------------------------------------------------------------
+# ----------------- Transforms ----------------------------------
+# ----------------------------------------------------------------
+
+
+@dataclass
+class ValidateCFConventions(beam.PTransform):
+    """
+    Transform to validate CF conventions
+    """
+
+    @staticmethod
+    def _get_dataset(store: zarr.storage.FSStore) -> xr.Dataset:
+        import xarray as xr
+
+        return xr.open_dataset(store, engine="zarr", chunks={}, use_cftime=True)
+
+    @staticmethod
+    def _retrieve_CF_axes(ds: xr.Dataset) -> dict[str, dict[str, str]]:
+        """Retrieve the CF dimensions from the dataset"""
+        import cf_xarray  # noqa
```

Comment:
move this at the top level since it is used in `_test_attributes` function ?

Reply:
To the top of the dataclass `ValidateCFConventions` or outside in with the other function?

Reply:
Top of the module as a global import...